src/Components/ProductCard/ProductCard.jsx

```diff
@@ -21,21 +23,36 @@ export function ProductCard({ products }) {
       <div className="product-list">
         <div className="Wishlist-container">
           <div className="card2-container wishlist-card">
-            <Link to="/wishlist">
-              <span>
-                <i className="far fa-heart fa-sm badge-style"></i>
-              </span>
-            </Link>
+            {wishlist.find((item) => item._id === products._id) ? (
```

Comment:
Inside of doing the logic directly inside jsx, you can create a function and do the logic there and call rhe function here. Simply like a component inside a component. When doing this wrap such component with useCallback Or useMemo to prevent unnecessary re rende